scenarioo-client/app/step/step.controller.js

```diff
@@ -464,35 +457,50 @@ function StepController($scope, $routeParams, $location, $route, StepResource, H
             });
     }
 
-    function initStorageKeys(){
-        if($scope.comparisonViewOptions[$scope.displaySideBySideViewKey] && $scope.comparisonViewOptions[$scope.displaySinglePageViewKey]){
-            setStorageKey($scope.displaySinglePageViewKey, false);
-        }
-        if (localStorageService.get($scope.displayComparisonScreenshotKey) === null){
-            setStorageKey($scope.displayComparisonScreenshotKey, false);
-        }
-    }
+     $scope.setComparisonView = function(viewId) {
+        $scope.comparisonViewOptions.viewId = viewId;
+        setLocalStorageValue('diffViewerStepComparisonViewId', viewId);
+    };
 
-    $scope.toggleStorageKey = function(storageKey) {
-        setStorageKey(storageKey, !$scope.comparisonViewOptions[storageKey]);
+    $scope.isComparisonView = function(viewId) {
+        return $scope.comparisonViewOptions.viewId === viewId;
     };
 
-    function setStorageKey(storageKey, value){
-        $scope.comparisonViewOptions[storageKey] = value;
-        localStorageService.set(storageKey, '' + value);
-    }
+    $scope.switchComparisonSingleScreenView = function() {
+        var viewId = $scope.isComparisonView('CurrentScreen') ? 'OtherScreen' : 'CurrentScreen';
+        $scope.setComparisonView(viewId);
+    };
 
-    $scope.setActiveView = function(storageKey) {
-         if(!$scope.comparisonViewOptions[storageKey]){
-            $scope.toggleStorageKey($scope.displaySideBySideViewKey);
-            $scope.toggleStorageKey($scope.displaySinglePageViewKey);
-        }
+    $scope.isComparisonChangesToBeHighlightedAvailable = function() {
+        return $scope.step.diffInfo.changeRate !== 0 && !$scope.step.diffInfo.isAdded;
     };
 
-    function getLocalStorageValue(storageKey){
+    $scope.isComparisonChangesHighlighted = function() {
+        // highlighting is turned on, and there are changes in this screenshot to be highlighted
+        return $scope.isComparisonChangesToBeHighlightedAvailable() && $scope.comparisonViewOptions.changesHighlighted;
+    };
+
+    $scope.toggleComparisonChangesHighlighted = function() {
+        $scope.comparisonViewOptions.changesHighlighted = !$scope.comparisonViewOptions.changesHighlighted;
+        setLocalStorageValue('diffViewerStepComparisonChangesHighlighted', $scope.comparisonViewOptions.changesHighlighted);
+    };
+
+    $scope.getComparisonViewHighlightChangesColor = function() {
+        return ConfigService.diffViewerDiffImageColor();
+    };
+
+    function getLocalStorageBool(storageKey){
```

Comment:
Can't we move it into the local storage service? So we could also unit test that one.

Reply:
yes, you are right on that - feel free to change - I have no time for it. I think some of that code I refactored here -
 especially concerning that access of properies from local storage - could have been done much better in the original code. Used me a lot of time because that was not designed in a way that was easy to change, I think I improved it already a lot. We can improve it even more of course!